device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceClientManagerSample.java

```diff
@@ -30,38 +39,89 @@
+                    break;
+                case "amqps_ws":
+                    argProtocol = IotHubClientProtocol.AMQPS_WS;
+                    break;
+                case "mqtt":
+                    argProtocol = IotHubClientProtocol.MQTT;
+                    break;
+                case "mqtt_ws":
+                    argProtocol = IotHubClientProtocol.MQTT_WS;
+                    break;
+                default:
+                    throw new IllegalArgumentException("[ERROR] Do not support protocol: [" + args[1] + "]");
+            }
+        }
+        else
+        {
+            argProtocol = IotHubClientProtocol.MQTT;
+            System.out.format("[DEFULT] Did not specify protocol. Default transport protocol to [%s]\n", argProtocol.name());
+            System.out.format(CMD_HELP, args.length);
+        }
+        System.out.format("Setup parameter: Protocol = [%s]\n", argProtocol.name());
+
+        int argNumRequest = (args.length > 2) ? Integer.parseInt(args[2]) : NUM_REQUESTS;
+        System.out.format("Setup parameter: Requests = [%d]\n", argNumRequest);
+        int argSleepDuration = (args.length > 3) ? Integer.parseInt(args[3]) : SLEEP_DURATION_IN_SECONDS;
+        System.out.format("Setup parameter: Sleep Duration = [%d]\n", argSleepDuration);
+        int argTimeout = (args.length > 4) ? Integer.parseInt(args[4]) : TIMEOUT_IN_MINUTES;
+        System.out.format("Setup parameter: Timeout = [%d]\n", argTimeout);
+
+        DeviceClient client = new DeviceClient(argDeviceConnectionString, argProtocol);
+        System.out.println("Successfully created an IoT Hub client.");
 
         deviceClientManager = new DeviceClientManager(client);
-        deviceClientManager.setOperationTimeout(DEVICE_OPERATION_TIMEOUT_IN_MINUTES);
+        deviceClientManager.setOperationTimeout(argTimeout);
 
         deviceClientManager.open();
-        log.debug("Opened connection to IoT Hub.");
+        System.out.println("Opened connection to IoT Hub.");
 
-        log.debug("Setting C2D message handler...");
+        System.out.println("Setting C2D message handler...");
         deviceClientManager.setMessageCallback(new SampleMessageReceiveCallback(), new Object());
 
-        log.debug("Start sending telemetry ...");
-        startSendingTelemetry();
+        System.out.println("Start sending telemetry ...");
+        startSendingTelemetry(argNumRequest, argSleepDuration);
 
         // close the connection
-        log.debug("Closing");
+        System.out.println("Closing");
         deviceClientManager.closeNow();
 
         if (! failedMessageListOnClose.isEmpty()) {
-            log.debug("List of messages that were cancelled on close: {}", failedMessageListOnClose.toString());
+            System.out.println("List of messages that were cancelled on close:");
+            System.out.println(failedMessageListOnClose.toString());
         }
 
-        log.debug("Shutting down...");
+        System.out.println("Shutting down...");
```

Comment:
I'd rather we kept these statements as log statements for this sample in particular. We will likely take a dependency on this sample in our longhaul tests, and having the logger gives us more flexibility than forcing the statements to System.out

Reply:
updated all to using log statements.